app/WebParsing/UtsgJsonParser.hs

```diff
@@ -1,26 +1,46 @@
 module WebParsing.UtsgJsonParser
      (getAllCourses,
-      getOrgs,
+      parseAllCourses,
       insertAllMeetings) where
 
 import Config (databasePath, orgApiUrl, timetableApiUrl)
 import Control.Monad.IO.Class (liftIO)
-import Data.Aeson (FromJSON (parseJSON), Object, Value (..), decode, (.!=), (.:?))
+import Data.Aeson (FromJSON (parseJSON), Object, Value (..), decode, decodeFileStrict, (.!=), (.:?))
 import Data.Aeson.Key (toText)
 import Data.Aeson.KeyMap as KM hiding (insert, map)
 import qualified Data.HashMap.Strict as HM
 import Data.Maybe (catMaybes)
 import qualified Data.Text as T
-import Database.Persist.Sqlite (SqlPersistM, insert, insertMany_, runSqlite, selectFirst, (==.))
+import Database.Persist
+import Database.Persist.Sqlite (SqlPersistM, runSqlite)
 import Database.Tables (Courses (..), EntityField (CoursesCode), MeetTime (..), Meeting (..),
-                        buildTimes)
+                        Times (..), buildTimes)
 import Network.HTTP.Conduit (simpleHttp)
 
+coursesJson :: FilePath
+coursesJson = "courses.json"
+
+parseAllCourses :: IO ()
+parseAllCourses = do
+    runSqlite databasePath parseCourses
+
+parseCourses :: SqlPersistM ()
+parseCourses = do
+    deleteWhere ([] :: [Filter Times])
+    deleteWhere ([] :: [Filter Meeting])
+    liftIO . print $ T.pack "parsing JSON data"
+    resp <- liftIO $ decodeFileStrict coursesJson
+    let coursesLst :: Maybe (HM.HashMap T.Text (Maybe DB)) = resp
+        courseData = maybe [] (map dbData . catMaybes . HM.elems) coursesLst
+        (_, sections) = unzip courseData
+        meetings = concat sections
+    mapM_ insertMeeting meetings
+
 -- | Parse all timetable data.
 getAllCourses :: IO ()
 getAllCourses = do
-    orgs <- getOrgs
-    runSqlite databasePath $ mapM_ insertAllMeetings orgs
+   orgs <- getOrgs
```

Comment:
@benxu3 there's still a diff showing here, so I don't think you completely reverted the change